### PR TITLE
Load config from table

### PR DESCRIPTION
**Pull request by wenzeslaus**
* Supports loading configuration from XLSX, ODS, and CSV in addition to the original YAML and JSON.
* Sheet and columns can be specified as part of the file name string (mostly useful for command line) or as separate parameters.
* Documentation updated to reflect the addition.
* Tests show the basic usages.
* No examples included. This waits for the final version of the spreadsheet with values filled in.
* There is a major code change included moving all config reading functions to a new file called inputs.py. This was needed because the table reading code added too many new lines and functions to simulation.py and it was also reusing code from scenarios.py.

### CODE REVIEW

**Comment by wenzeslaus** [comment]
Description updated to include the main changes.

File: docs/run.md [hunk 1]
```diff
@@ -2,16 +2,36 @@
 
 The two basic ways to run the simulation are a Python package and command line interface. Both interfaces take simulation parameters as a configuration file and several other user inputs as function arguments or command line arguments.
```
Comment (kellynm) [inline docs/run.md:3]: This file needs text line wrapping.

File: docs/cli.md [hunk 2]
```diff
@@ -5,18 +5,25 @@
...
 dependencies and it will work in any directory. 
 
+For spreadsheet (tabular) formats, you need specify `sheet`, `key_column`,
+and `value_column` together with the file name in the `--config-file`
+separated by `::` from the file name as key-value pairs
+where key and value are separated by `=` and individual keys are separated by `,`.
+
 ## Example
 
 Direct run of the simulation with 1000 consignments repeated 20 times:
```
Comment (kellynm) [inline docs/cli.md:29]: Could you add one additional example here that uses a tabular config with values for `sheet`, `key_column`,
and `value_column`?

**Comment by kellynm** [comment]
I think this all looks good - thank you for implementing this!

What do you think about putting a copy of the user-friendly xlsx config in the root, with config.yaml? And maybe mention in run.md that these two files are in the repo and can be used as a template? I'm thinking we will use the user-friendly config for one of the examples in the tutorial notebook, to be written at a later date.

File: docs/cli.md (hunk 2 above)
Comment (wenzeslaus) [reply docs/cli.md:33]: Done for CSV (columns as numbers although the letters work too) and XLSX (includes sheet).

File: docs/run.md (hunk 1 above)
Comment (wenzeslaus) [reply docs/run.md]: Done for the whole file, 80 chars. We don't have anything in place to check for it. Maybe I should add it.

**Comment by wenzeslaus** [comment]
Thanks for looking at it @kellynm!

> What do you think about putting a copy of the user-friendly xlsx config in the root, with config.yaml?

Yes, definitively. Not to the root, the YAML file should not be there, but in examples. I guess I forgot to change it (#138). Either in one of the existing directories in examples or in a separate sub-directory directory such as examples/config or examples/templates or top-level templates... In any case, I want to leave this for another PR, so we can better focus on what we want in the example and do the whole example and/or template at once.

> And maybe mention in run.md that these two files are in the repo and can be used as a template?

Agreed, now tracking all this in #147. We do whatever is best for the new XLSX and deal with the existing YAML file accordingly.

> I'm thinking we will use the user-friendly config for one of the examples in the tutorial notebook, to be written at a later date.

The current example we have in GDrive and I copied to the tests does not have any values in it. For what I needed to test (reading config in a fancy spread form), it is enough, but I'm leaving actual example for later.

**Pull request merged**